samplePDF/samplePDFFDBase.cpp

```diff
@@ -1428,96 +1454,166 @@ double samplePDFFDBase::GetEventWeight(int iSample, int iEntry) {
+  if (iSample < 0 || iSample >= nSamples) {
+    MACH3LOG_ERROR("Invalid iSample index in InitialiseSingleFDMCObject");
+    MACH3LOG_ERROR("Index given is {} and only {} samples found", iSample, nSamples);
+    throw MaCh3Exception(__FILE__, __LINE__);
   }
+  
+  MCSamples[iSample] = fdmc_base();
+  fdmc_base *fdobj = &MCSamples[iSample];
+  
+  fdobj->nEvents = nEvents_;
+  fdobj->nutype = -9;
+  fdobj->oscnutype = -9;
+  fdobj->signal = false;
+  fdobj->Unity = 1.;
+  fdobj->Unity_Int = 1.;
+  
+  fdobj->x_var = new const double*[fdobj->nEvents];
+  fdobj->y_var = new const double*[fdobj->nEvents];
+  fdobj->rw_etru = new const double*[fdobj->nEvents];
+  fdobj->XBin = new int[fdobj->nEvents];
+  fdobj->YBin = new int[fdobj->nEvents];    
+  fdobj->NomXBin = new int[fdobj->nEvents];
+  fdobj->NomYBin = new int[fdobj->nEvents];
+  fdobj->XBin = new int [fdobj->nEvents];
+  fdobj->YBin = new int [fdobj->nEvents];;	 
+  fdobj->rw_lower_xbinedge = new double [fdobj->nEvents];
+  fdobj->rw_lower_lower_xbinedge = new double [fdobj->nEvents];
+  fdobj->rw_upper_xbinedge = new double [fdobj->nEvents];
+  fdobj->rw_upper_upper_xbinedge = new double [fdobj->nEvents];
+  fdobj->mode = new int*[fdobj->nEvents];
+  fdobj->nxsec_norm_pointers = new int[fdobj->nEvents];
+  fdobj->xsec_norm_pointers = new const double**[fdobj->nEvents];
+  fdobj->xsec_norms_bins = new std::list< int >[fdobj->nEvents];
+  fdobj->xsec_w = new double[fdobj->nEvents];
+  fdobj->isNC = new bool[fdobj->nEvents];
+  fdobj->nxsec_spline_pointers = new int[fdobj->nEvents];
+  fdobj->xsec_spline_pointers = new const double**[fdobj->nEvents];
+  fdobj->ntotal_weight_pointers = new int[fdobj->nEvents];
+  fdobj->total_weight_pointers = new const double**[fdobj->nEvents];
+  fdobj->Target = new int*[fdobj->nEvents];
+  fdobj->osc_w_pointer = new const _float_*[fdobj->nEvents];
+  //fdobj->rw_truecz = new const double*[fdobj->nEvents];
+  
+  for(int iEvent = 0 ;iEvent < fdobj->nEvents ; ++iEvent){
+    fdobj->rw_etru[iEvent] = &fdobj->Unity;
+    //fdobj->rw_truecz[iEvent] = &fdobj->Unity;
+    fdobj->mode[iEvent] = &fdobj->Unity_Int;
+    fdobj->Target[iEvent] = 0;
+    fdobj->NomXBin[iEvent] = -1;
+    fdobj->NomYBin[iEvent] = -1;
+    fdobj->XBin[iEvent] = -1;
+    fdobj->YBin[iEvent] = -1;
+    fdobj->rw_lower_xbinedge[iEvent] = -1;
+    fdobj->rw_lower_lower_xbinedge[iEvent] = -1;
+    fdobj->rw_upper_xbinedge[iEvent] = -1;
+    fdobj->rw_upper_upper_xbinedge[iEvent] = -1;
+    fdobj->xsec_w[iEvent] = 1.0;
+    fdobj->isNC[iEvent] = false;
+    fdobj->SampleDetID = -1;
+    #ifdef _LOW_MEMORY_STRUCTS_
+    fdobj->osc_w_pointer[iEvent] = &(fdobj->Unity_F);
+    #else
+    fdobj->osc_w_pointer[iEvent] = &(fdobj->Unity);
+    #endif
+
+    fdobj->x_var[iEvent] = &fdobj->Unity;
+    fdobj->y_var[iEvent] = &fdobj->Unity;
+  }
+
 }
-#endif
 
+void samplePDFFDBase::InitialiseSplineObject() {
+  std::vector<std::string> spline_filepaths;
```

Comment:
can this code not be moved to the constructor of the spline object? or would there be too many arguments to pass? Can the arguments be wrapped up in a 'config struct' that can be passed to spline objects to initialise themselves?